sampleExtension/optionsPage/options.js

```diff
@@ -2,66 +2,83 @@
-        interceptionCounter = output.tds_interceptCounter;
-        if (!chrome.runtime.error) {
+initOptionsPage = function() {
+    chrome.storage.sync.get(["tds_blacklist", "tds_interceptCounter"], function(output) {
+        if (handleRuntimeError()) {
+            setLocalVariables(output);
             html_intCnt.text(interceptionCounter);
+            setHtmlBlacklist(links);
         }
     });
 };
 
+setLocalVariables = function(output) {
+    links = output.tds_blacklist;
+    interceptionCounter = output.tds_interceptCounter;
+};
+
+setHtmlBlacklist = function(list) {
+    $.each(list, function(key, value) {
+        html_blacklist.append($("<option></option>").text(value));
+    });
+};
+
+
+/* -------------------- Manipulate storage and variables ------------------- */
+
+updateBackgroundPage = function() {
+    var bg = chrome.extension.getBackgroundPage();
+    bg.updateBlockedSites(bg.replaceListener);
+};
+
+updateStorageBlacklist = function() {
+    setStorageBlacklistWithCallback(links, updateBackgroundPage);
+};
+
+removeFromAll = function(html_item) {
+    removeFromLinks(html_item);
+    //update html_blacklist
+    html_item.remove();
+    updateStorageBlacklist();
+};
+
+removeFromLinks = function(html_item) {
+    var urlkey = links.indexOf(html_item.val());
+    links.splice(urlkey, 1);
+};
+
+addToAll = function(item) {
+    var stringVal = "*://"+item+"/*";
+    links.push(stringVal);
+    html_blacklist.append($("<option></option>").text(stringVal));
+    updateStorageBlacklist();
+};
+
 
 /* -------------------- Logic for the html buttons -------------------- */
 
 saveButtonClick = function() {
     var newurl = html_txtFld.val();
-    // Add the url to the sync storage.
-    links.push("*://"+newurl+"/*");
-    chrome.storage.sync.set({"tds_blacklist" : links }, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-        //Append element to the html blacklist
-        html_blacklist.append($("<option></option>").text("*://"+newurl+"/*"));
-        // Empty the input box.
-        html_txtFld.val('');
-        var bg = chrome.extension.getBackgroundPage();
-        bg.updateBlockedSites(bg.replaceListener);
-    });
+    addToAll(newurl);
+    html_txtFld.val('');
 };
 
 deleteButtonClick = function() {
-    var urltodelete = $("#blacklistedSites").find('option:selected');
-    var urlkey = links.indexOf(urltodelete.val());
-    // Remove the url from the sync storage.
-    links.splice(urlkey, 1);
-    chrome.storage.sync.set({"tds_blacklist" : links}, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-        // Remove the url from the list.
-        urltodelete.remove();
-        var bg = chrome.extension.getBackgroundPage();
-        bg.updateBlockedSites(bg.replaceListener);
-    });
+    var urlToDelete = html_blacklist.find('option:selected');
+    removeFromAll(urlToDelete)
```

Comment:
Mark and I think that it would be better to have 2 functions : removeFromHTML and removeFromStorage that would be called one after another, than having 2 functionalities wrapped in the same function :)